tests/Replication2/Serialization/LogStatusTest.cpp

```diff
@@ -0,0 +1,297 @@
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Alexandru Petenchea
+////////////////////////////////////////////////////////////////////////////////
+
+#include "Basics/VelocyPackHelper.h"
+#include "velocypack/Builder.h"
+#include "velocypack/velocypack-aliases.h"
+
+#include <gtest/gtest.h>
+
+#include "Replication2/ReplicatedLog/LogStatus.h"
+#include "Replication2/ReplicatedLog/types.h"
+
+using namespace arangodb;
+using namespace arangodb::replication2;
+using namespace arangodb::replication2::replicated_log;
+using namespace arangodb::basics;
+using namespace std::literals::chrono_literals;
+
+static inline auto vpackFromJsonString(char const* c) -> VPackBuffer<uint8_t> {
+  VPackOptions options;
+  options.checkAttributeUniqueness = true;
+  VPackParser parser(&options);
+  parser.parse(c);
+
+  std::shared_ptr<VPackBuilder> builder = parser.steal();
+  std::shared_ptr<VPackBuffer<uint8_t>> buffer = builder->steal();
+  VPackBuffer<uint8_t> b = std::move(*buffer);
+  return b;
+}
+
+static inline auto operator"" _vpack(const char* json, size_t) {
+  return vpackFromJsonString(json);
+}
+
+namespace arangodb::replication2::replicated_log {
+[[maybe_unused]] static inline auto operator==(LogStatistics const& left,
+                                               LogStatistics const& right) {
+  return left.spearHead == right.spearHead &&
+         left.commitIndex == right.commitIndex &&
+         left.firstIndex == right.firstIndex;
+}
+
+[[maybe_unused]] static inline auto operator==(CommitFailReason const& left,
+                                               CommitFailReason const& right) {
+  return left.value.index() == right.value.index();
+}
+
+[[maybe_unused]] static inline auto operator==(FollowerStatistics const& left,
+                                               FollowerStatistics const& right) {
+  return left.lastErrorReason == right.lastErrorReason &&
+         left.lastRequestLatencyMS == right.lastRequestLatencyMS &&
+         left.internalState.value.index() == right.internalState.value.index();
+}
+
+[[maybe_unused]] static inline auto operator==(LeaderStatus const& left,
+                                               LeaderStatus const& right) {
+  bool result = left.local == right.local &&
+                left.term == right.term &&
+                left.largestCommonIndex == right.largestCommonIndex &&
+                left.commitLagMS == right.commitLagMS &&
+                left.lastCommitStatus == right.lastCommitStatus
+                && left.follower.size() == right.follower.size();
+  if (!result) {
+    return false;
+  }
+  for (auto const& [participantId, followerStatistics] : left.follower) {
+    auto search = right.follower.find(participantId);
+    if (search == right.follower.end() || !(search->second == followerStatistics)) {
+      result = false;
+      break;
+    }
+  }
+  return result;
+}
```

Comment:
It might be handy to make these operators publicly available, i.e. add them as friends to the classes. (Similar to TermIndexPair). And don't forget to add `!=` as `!(lhs == rhs)`.

And as soon as C++20 is available, we can just delete the implementation and use `default`.